test/common/schemaform/fields/ArrayField.unit.spec.jsx

```diff
@@ -144,55 +229,55 @@ describe('Schemaform <ArrayField>', () => {
       );
     });
     it('edit', () => {
-      expect(tree.everySubTree('SchemaField').length).to.equal(1);
+      expect(tree.everySubTree('SchemaField').length).to.equal(0);
 
       tree.getMountedInstance().handleEdit(0);
 
-      expect(tree.everySubTree('SchemaField').length).to.equal(2);
+      expect(tree.everySubTree('SchemaField').length).to.equal(1);
     });
     it('update when valid', () => {
       tree.getMountedInstance().handleEdit(0);
 
-      expect(tree.everySubTree('SchemaField').length).to.equal(2);
+      expect(tree.everySubTree('SchemaField').length).to.equal(1);
 
       tree.getMountedInstance().handleUpdate(0);
 
-      expect(tree.everySubTree('SchemaField').length).to.equal(1);
+      expect(tree.everySubTree('SchemaField').length).to.equal(0);
     });
     it('not update when invalid', () => {
       tree.getMountedInstance().handleEdit(0);
 
-      expect(tree.everySubTree('SchemaField').length).to.equal(2);
+      expect(tree.everySubTree('SchemaField').length).to.equal(1);
 
-      errorSchema[0] = { __errors: ['Testing'] };
+      fullErrorSchema[0] = { __errors: ['Testing'] };
 
       tree.getMountedInstance().handleUpdate(0);
 
-      expect(tree.everySubTree('SchemaField').length).to.equal(2);
+      expect(tree.everySubTree('SchemaField').length).to.equal(1);
     });
     it('add', () => {
-      expect(tree.everySubTree('SchemaField').length).to.equal(1);
+      expect(tree.everySubTree('SchemaField').length).to.equal(0);
 
       tree.getMountedInstance().handleAdd();
 
       expect(onChange.firstCall.args[0].length).to.equal(3);
-      expect(tree.getMountedInstance().state.editing[2]).to.be.false;
```

Comment:
The new `expect` looks good, but this one confuses me, and I find that concerning. In this old test, this is my understanding:
1) We have two items in the array
2) The first item is rendered as not editing (`viewField`)
3) The second item is rendered as editing (because it's the last item in the array)
4) When we hit the add button, it puts the second item into a view state (not editing) and adds a third
5) This newly-added third item should be rendered in the edit state

So...why did we expect the edit state of the third item to be false? @jbalboni?

Reply:
One thing that might be confusing here is that this no longer called `editing` but `itemModes`, and the new item's `itemMode` is now being tested as being `adding`, in case that helps to clarify what the change is here.